### PR TITLE
CEDS-2329 - Migrate 'Holder of Authorisation' to Gov Design System

### PR DESCRIPTION
**Pull request by iservport**

### CODE REVIEW

File: app/views/declaration/declaration_holder.scala.html [hunk 1]
```diff
@@ -20,62 +20,121 @@
...
-@main_template(
-    title = Title("supplementary.declarationHolder.title", "supplementary.summary.parties.header")
-) {
+@addButton = @{
+govukButton(Button(
+    classes = "govuk-button--secondary",
+    content = HtmlContent(HtmlFormat.fill(
+        immutable.Seq(
+            Html(s"""<span aria-hidden="true">${messages("site.add")}</span>"""),
+            Html(s"""<span class="govuk-visually-hidden">${messages("supplementary.additionalActors.add.hint")}</span>""")
```
Comment (PauloPortugal) [inline app/views/declaration/declaration_holder.scala.html:57]: The Hidden message is incorrect

File: app/views/declaration/declaration_holder.scala.html [hunk 2]
```diff
@@ -20,62 +20,121 @@
...
+        )
+    )),
+    name = Some("Add"),
+    attributes = Map("id" -> "add")
+))
+}
 
-    @components.back_link(Navigator.backLink(DeclarationHolder, mode), mode)
+@removeButton(value: Option[String]) = @{
```
Comment (PauloPortugal) [inline app/views/declaration/declaration_holder.scala.html:65]: We have a `removeButton` partial, please use that one ;)

File: app/views/declaration/declaration_holder.scala.html [hunk 3]
```diff
@@ -20,62 +20,121 @@
...
+)
 
 @(mode: Mode, form: Form[DeclarationHolder], holders: Seq[DeclarationHolder])(implicit request: JourneyRequest[_], messages: Messages)
 
 @tariffBody = { @Html(messages("declaration.type.holderOfAuthorisationSummaryText", components.details_content_link())) }
 
-@main_template(
-    title = Title("supplementary.declarationHolder.title", "supplementary.summary.parties.header")
-) {
+@addButton = @{
```
Comment (PauloPortugal) [inline app/views/declaration/declaration_holder.scala.html:51]: we have a `addButton` partial, please use that one via DI on the view.

File: app/views/declaration/declaration_holder.scala.html (hunk 1 above)
Comment (iservport) [reply app/views/declaration/declaration_holder.scala.html:57]: Bad copy-paste...

File: app/views/declaration/declaration_holder.scala.html (hunk 3 above)
Comment (SteveSugden) [reply app/views/declaration/declaration_holder.scala.html:51]: See procedure_codes view

File: app/views/declaration/declaration_holder.scala.html (hunk 2 above)
Comment (SteveSugden) [reply app/views/declaration/declaration_holder.scala.html:65]: See procedure_codes view

File: app/views/declaration/declaration_holder.scala.html [hunk 4]
```diff
@@ -20,62 +20,121 @@
...
-    )
+    @formHelper(action = DeclarationHolderController.submitHoldersOfAuthorisation(mode), 'autoComplete -> "off") {
+        @errorSummary(form.errors)
 
-    <p>@messages("supplementary.declarationHolder.title.hint")</p>
+        @sectionHeader(messages("supplementary.summary.parties.header"))
 
-    @helper.form(DeclarationHolderController.submitHoldersOfAuthorisation(mode), 'autoComplete -> "off") {
-        @helper.CSRF.formField
+        @pageTitle(messages("supplementary.declarationHolder.title"))
```
Comment (SteveSugden) [inline app/views/declaration/declaration_holder.scala.html:114]: Can we take this opportunity to rename `supplementary.declarationHolder...` messages to `declaration.declarationHolder...`

File: test/views/declaration/DeclarationHolderViewSpec.scala [hunk 5]
```diff
@@ -82,13 +82,11 @@ class DeclarationHolderViewSpec extends UnitViewSpec with CommonMessages with St
 
       "display empty input with label for EORI" in {
 
-        document.getElementById("eori-label").text() mustBe messages("supplementary.declarationHolder.eori")
+        document.getElementsByClass("govuk-label govuk-label--m").get(1).text() mustBe messages("supplementary.declarationHolder.eori")
```
Comment (SteveSugden) [inline test/views/declaration/DeclarationHolderViewSpec.scala:85]: That looks fragile way of getting the element.   I think that is an input field label you are trying to find.
`document.getElementsByAttributeValue("for", "eori").text()` should work better and as an added bonus tests that the label is correctly "for" the input field.

Comment (iservport) [reply test/views/declaration/DeclarationHolderViewSpec.scala:85]: Good hint

File: conf/messages.en [hunk 6]
```diff
@@ -433,16 +433,16 @@ saved.declarations.remove.title = Are you sure you want to remove this declarati
...
+declaration.declarationHolder.title = What authorisation is required for this export?
+declaration.declarationHolder.title.hint = This will depend on the type of procedure code used for this export. If no authorisation is needed, leave this page blank and continue.
+declaration.declarationHolder.eori = EORI number of the authorised party
+declaration.declarationHolder.authorisationCode = Authorisation code
+declaration.declarationHolder.authorisationCode.hint = A code up to 4 characters for the type of authorisation.
+declaration.declarationHolder.authorisationCode.invalid = Enter a valid authorisation code
+declaration.declarationHolder.authorisationCode.empty = Enter a valid authorisation code
+declaration.declarationHolders.maximumAmount.error = Maximum amount of holders added
+declaration.declarationHolders.duplicated = Holder's details already added
+declaration.declarationHolders.add.hint = Add an authorised party
```
Comment (SteveSugden) [inline conf/messages.en:445]: 👍 👍

**Pull request merged**